packages/vscode-extension/src/project/project.ts

```diff
@@ -201,13 +200,21 @@ export class Project
     };
 
     const devices = await this.deviceManager.listAllDevices();
-    if (!selectInitialDevice(devices)) {
-      const listener = (newDevices: DeviceInfo[]) => {
-        if (selectInitialDevice(newDevices)) {
-          this.deviceManager.removeListener("devicesChanged", listener);
-        }
-      };
-      this.deviceManager.addListener("devicesChanged", listener);
+    const selectInitialDevicePromise = selectInitialDevice(devices);
+
+    const listener = async (newDevices: DeviceInfo[]) => {
+      if (await selectInitialDevicePromise) {
+        this.deviceManager.removeListener("devicesChanged", listener);
+        return;
+      }
+      if (await selectInitialDevice(newDevices)) {
+        this.deviceManager.removeListener("devicesChanged", listener);
+      }
+    };
+    this.deviceManager.addListener("devicesChanged", listener);
+
+    if (await selectInitialDevicePromise) {
+      this.deviceManager.removeListener("devicesChanged", listener);
```

Comment:
This code doesn't do the same thing as the one on the left. In general I'm most afraid of the changes in awaiting that you're adding here. Not saying we shouldn't have them, but they require some explanation in the PR description, such that we can understand why we're changing the await flow.

This part is the most complex change and it introduces some unusual pattern (that we, albeit, use in other places but typically with some explanation) where we create promise and then await inside of an arrow function.

I think that given this method already awaits `selectDevice` (which wasn't the case before this change) , we can just add the listener before calling `selectInitialDevice` ?